docs: blogpost for cli v2 (#30876)
supabase/supabase: apps/www/_blog/2024-12-04-cli-v2-config-as-code.mdx

---
title: 'Supabase CLI v2: Config as Code'
description: Edge functions can be used for workloads outside the request-response lifecycle
author: qiao,avallete
image: launch-week-13/day-2-functions-background-tasks/og.jpg
thumb: launch-week-13/day-2-functions-background-tasks/thumb.jpg
categories:
  - product
tags:
  - launch-week
date: '2024-12-04'
toc_depth: 3
launchweek: '13'
---

> The Supabase CLI v2 was released today, adding support for Configuration as Code. This means you can commit the configuration for all of your Projects and Branches into version control (like git) for reproducible environments for your entire team.

Reviewer: ```suggestion
We have released Supabase CLI v2 today, adding support for Configuration as Code. 

This means you can commit the configuration for all of your Projects and Branches into version control (like git) for reproducible environments for your entire team.
```

## Using the CLI in CI/CD pipelines

> The Supabase CLI started out as a way to bootstrap the entire Supabase stack on your local machine. The CLI uses exactly the same infra as our hosted platform, giving you unlimited Supabase projects for local testing and offline usage.

Reviewer: ```suggestion
The Supabase CLI started as a way to bootstrap the entire Supabase stack on your local machine. It uses exactly the same infra as our hosted platform, giving you unlimited Supabase projects for local testing and offline usage.
```

> Over the last 2 years the CLI has grown to more than 180,000 weekly installs. Close to 85% of the downloads come from Continuous Integration/Deployment environments like GitHub Actions. Some of the popular CI/CD use cases include migrating production database, deploying functions, and running pgTAP tests. With this in mind, we started focusing on the CLI as a deployment tool for the v2 release.

Reviewer: ```suggestion
In the last 2 years, the CLI has grown to more than 180,000 weekly installs. Nearly 85% of these come from Continuous Integration/Deployment environments like GitHub Actions. Some of the popular CI/CD use cases include migrating production databases, deploying functions, and running pgTAP tests. With this in mind, we started focusing on the CLI as a deployment tool for the v2 release.
```

- **Version Control**: Configurations are stored in repositories like Git, enabling collaboration and history tracking.
- **Reproducibility**: You can recreate environments from code, avoiding manual drift.
- **Automation**: Tools apply configurations automatically, reducing human error.

Our CLI’s Configuration as Code feature is an opinionated setup using a human readable `config.toml` file.

You can make deployments consistent and repeatable by promoting Edge Functions, Storage objects, and other services from preview environments to staging and production.

To demonstrate this workflow, let’s use the `supabase.com` website as an example. It’s hosted on Vercel with [Supabase Branching](/docs/guides/deployment/branching) enabled for development. If you are not using Branching, a similar setup can be achieved using GitHub Actions.

<aside>

💡 Before changing any project configuration, it’s a good idea to verify that your remote config has not drifted. This can be done by running `supabase link` command locally which diffs your entire local `config.toml` with your remote project settings.

> </aside>

Reviewer: Let's make this an <Admonition> like https://supabase.com/blog/supabase-bootstrap ?

### Managing Auth Config

> Since our frontend uses Vercel Preview, we want to configure the Auth service of Supabase branches to support login for any Vercel preview URL. This can be achieved by declaring a wildcard for the `additional_redirect_urls` in auth config:

Reviewer: ```suggestion
We use Vercel Previews for our frontend. To configure the Auth service of Supabase branches to support login for any Vercel preview URL, we declare a wildcard for the `additional_redirect_urls` in auth config:
```

```toml
[auth]
additional_redirect_urls = [
  "https://*-supabase.vercel.app/*/*",
  "https://supabase.com/*/*",
  "http://localhost:3000/*/*",
]
```

View the [Auth config](/docs/guides/local-development/cli/config#auth-config) docs.

### Managing Edge Functions

The Supabase website uses several [Edge Functions](https://github.com/supabase/supabase/tree/master/supabase/functions) for AI docs, search embeddings, and image generation for launch week tickets. To configure automatic deployment of `search-embeddings` function, we add the following block to `config.toml`:

```toml
[functions.search-embeddings]
verify_jwt = false
```

If you are using a monorepo (like the [@supabase/supabase](https://github.com/supabase/supabase) Github repository), you may also want to customize the paths to your function’s entrypoint and import map files. This is especially useful for code sharing between your frontend application and Edge Functions.

> <aside>

Reviewer: admonition
